### PR TITLE
Deprecate System role in EACL #616

### PR DESCRIPTION
**Pull request by vvarg229**
Removed the System role in EACL and removed the tests related to this role. The System EACL role leads to a broken container and nothing else and will be removed in the issue https://github.com/nspcc-dev/neofs-node/issues/2531

Tests result:
https://http.t5.fs.neo.org/86C4P6uJC7gb5n3KkwEGpXRfdczubXyRNW5N9KeJRW73/332-1693924040/index.html#

### CODE REVIEW

**Review by carpawell — COMMENTED** [review]
Will we add some tests about _future_ system changes? I mean refusing system changes will be added in the `neofs-cli` and IR. Also, some `--force` may be added. So some issues about that?

**Comment by vvarg229** [comment]
> * we just deprecate the role, we should continue to test it
These tests are the cause of cluster breakage, which affects other tests.

> * we forbid it, then we should test that it's denied by the system (try and expect failure)
I agree, it's a good idea to add tests like this.
@roman-khimov what do you think?

**Comment by vvarg229** [comment]
> Will we add some tests about _future_ system changes?
I think we should add tests for not being able to change System role using EACL. But it seems to me that this should be done after the issue https://github.com/nspcc-dev/neofs-node/issues/2531 is resolved.

> I mean refusing system changes will be added in the `neofs-cli` and IR. Also, some `--force` may be added. So some issues about that?
Let's discuss this in the https://github.com/nspcc-dev/neofs-node/issues/2531 issue?

File: pytest_tests/testsuites/acl/test_eacl.py [hunk 1]
```diff
@@ -252,436 +252,6 @@ def test_extended_acl_deny_replication(
                 storage_nodes,
             )
 
-    @allure.title("Testcase to validate NeoFS system operations with extended ACL")
```
Comment (roman-khimov) [inline pytest_tests/testsuites/acl/test_eacl.py:255]: I don't see anything similar for basic ACL, is that the case?

File: pytest_tests/testsuites/acl/test_eacl.py [hunk 2]
```diff
@@ -252,436 +252,6 @@ def test_extended_acl_deny_replication(
...
-    def test_extended_actions_system(self, wallets, eacl_container_with_objects):
-        user_wallet = wallets.get_wallet()
-        ir_wallet, storage_wallet = wallets.get_wallets_list(role=EACLRole.SYSTEM)[:2]
```
Comment (roman-khimov) [inline pytest_tests/testsuites/acl/test_eacl.py:258]: This, btw is a really bad style.

File: pytest_tests/testsuites/acl/test_eacl.py [hunk 3]
```diff
@@ -252,436 +252,6 @@ def test_extended_acl_deny_replication(
...
-                wallet_config=ir_wallet.config_path,
-            )
-            assert can_put_object(
-                storage_wallet.wallet_path,
-                cid,
-                file_path,
-                shell=self.shell,
-                cluster=self.cluster,
-                wallet_config=storage_wallet.config_path,
-            )
```
Comment (roman-khimov) [inline pytest_tests/testsuites/acl/test_eacl.py:279]: For example, this pair perfectly makes sense as a test for basic ACL, SN should be able to PUT, while IR should not.

File: pytest_tests/testsuites/acl/test_eacl.py [hunk 4]
```diff
@@ -252,436 +252,6 @@ def test_extended_acl_deny_replication(
...
-            assert can_get_object(
-                storage_wallet.wallet_path,
-                cid,
-                object_oids[0],
-                file_path,
-                shell=self.shell,
-                cluster=self.cluster,
-                wallet_config=storage_wallet.config_path,
-            )
```
Comment (roman-khimov) [inline pytest_tests/testsuites/acl/test_eacl.py:298]: This doesn't make much sense.

File: pytest_tests/testsuites/acl/test_eacl.py [hunk 5]
```diff
@@ -252,436 +252,6 @@ def test_extended_acl_deny_replication(
...
-                wallet_config=ir_wallet.config_path,
-            )
-            assert can_get_head_object(
-                storage_wallet.wallet_path,
-                cid,
-                object_oids[0],
-                shell=self.shell,
-                endpoint=endpoint,
-                wallet_config=storage_wallet.config_path,
-            )
```
Comment (roman-khimov) [inline pytest_tests/testsuites/acl/test_eacl.py:315]: Like this

File: pytest_tests/testsuites/acl/test_eacl.py [hunk 6]
```diff
@@ -252,436 +252,6 @@ def test_extended_acl_deny_replication(
...
-                wallet_config=ir_wallet.config_path,
-            )
-            assert can_search_object(
-                storage_wallet.wallet_path,
-                cid,
-                shell=self.shell,
-                endpoint=endpoint,
-                oid=object_oids[0],
-                wallet_config=storage_wallet.config_path,
-            )
```
Comment (roman-khimov) [inline pytest_tests/testsuites/acl/test_eacl.py:332]: And this.

File: pytest_tests/testsuites/acl/test_eacl.py [hunk 7]
```diff
@@ -252,436 +252,6 @@ def test_extended_acl_deny_replication(
...
-                )
-            with pytest.raises(AssertionError):
-                assert can_get_range_of_object(
-                    wallet=storage_wallet.wallet_path,
-                    cid=cid,
-                    oid=object_oids[0],
-                    shell=self.shell,
-                    endpoint=endpoint,
-                    wallet_config=storage_wallet.config_path,
-                )
```
Comment (roman-khimov) [inline pytest_tests/testsuites/acl/test_eacl.py:351]: And this

File: pytest_tests/testsuites/acl/test_eacl.py [hunk 8]
```diff
@@ -252,436 +252,6 @@ def test_extended_acl_deny_replication(
...
-            )
-
-            assert can_get_range_hash_of_object(
-                wallet=storage_wallet.wallet_path,
-                cid=cid,
-                oid=object_oids[0],
-                shell=self.shell,
-                endpoint=endpoint,
-                wallet_config=storage_wallet.config_path,
-            )
```
Comment (roman-khimov) [inline pytest_tests/testsuites/acl/test_eacl.py:369]: And this

File: pytest_tests/testsuites/acl/test_eacl.py [hunk 9]
```diff
@@ -252,436 +252,6 @@ def test_extended_acl_deny_replication(
...
-                )
-            with pytest.raises(AssertionError):
-                assert can_delete_object(
-                    wallet=storage_wallet.wallet_path,
-                    cid=cid,
-                    oid=object_oids[0],
-                    shell=self.shell,
-                    endpoint=endpoint,
-                    wallet_config=storage_wallet.config_path,
-                )
```
Comment (roman-khimov) [inline pytest_tests/testsuites/acl/test_eacl.py:388]: While deletion test has some meaning, none of them should be able to do that.

**Review by roman-khimov — CHANGES_REQUESTED** [review]

**Comment by roman-khimov** [comment]
When will this be updated?
https://http.fs.neo.org/HXSaMJXk2g8C14ht8HSi7BBaiYZ1HeWh2xnWPGQCg4H6/722-1694457866/index.html#suites/69b447004f3c682f67955ec900a09cf0/d419bcbd5433e713/

File: pytest_tests/testsuites/acl/test_eacl.py (hunk 1 above)
Comment (vvarg229) [reply pytest_tests/testsuites/acl/test_eacl.py:255]: > I don't see anything similar for basic ACL, is that the case?

I don't quite understand this comment. How is basic ACL and these changes related?

Comment (vvarg229) [reply pytest_tests/testsuites/acl/test_eacl.py:255]: > I don't see anything similar for basic ACL, is that the case?

I don't quite understand this comment. How is basic ACL and these changes related?

Comment (roman-khimov) [reply pytest_tests/testsuites/acl/test_eacl.py:255]: They're not directly related, the question is do we have similar tests for basic ACL? If we're not --- let's have an issue for it.

Comment (vvarg229) [reply pytest_tests/testsuites/acl/test_eacl.py:255]: > They're not directly related, the question is do we have similar tests for basic ACL? If we're not --- let's have an issue for it.

https://github.com/nspcc-dev/neofs-testcases/issues/636

Comment (roman-khimov) [reply pytest_tests/testsuites/acl/test_eacl.py:255]: #634. So you want to merge it as is? Frankly, I've expected it to be solved in a day. If that takes too long, OK, let's move on with stripped test and a separate issue.

Comment (vvarg229) [reply pytest_tests/testsuites/acl/test_eacl.py:255]: > #634. So you want to merge it as is? Frankly, I've expected it to be solved in a day. If that takes too long, OK, let's move on with stripped test and a separate issue.

No, I don't want to merge it as is.
But it seems to me that tests for basic acl are a separate task.

I have a question - should basic acl allow changing the system role?
Wouldn't this lead to the same bugs that we fix by removing this functionality from eacl?

And here I will leave the tests that check negative scenarios with system eacl.

Comment (roman-khimov) [reply pytest_tests/testsuites/acl/test_eacl.py:255]: That's the beauty of basic ACL, it was always implemented in a way forbidding messing with system bits. They can only be set to what they need to be set according to the spec.

Comment (vvarg229) [reply pytest_tests/testsuites/acl/test_eacl.py:255]: > > #634. So you want to merge it as is? Frankly, I've expected it to be solved in a day. If that takes too long, OK, let's move on with stripped test and a separate issue.
> 
> No, I don't want to merge it as is. But it seems to me that tests for basic acl are a separate task.
> 
> I have a question - should basic acl allow changing the system role? Wouldn't this lead to the same bugs that we fix by removing this functionality from eacl?
> 
> And here I will leave the tests that check negative scenarios with system eacl.

Fixed

File: pytest_tests/testsuites/acl/test_eacl.py (hunk 3 above)
Comment (vvarg229) [reply pytest_tests/testsuites/acl/test_eacl.py:279]: > For example, this pair perfectly makes sense as a test for basic ACL, SN should be able to PUT, while IR should not.

Fixed

File: pytest_tests/testsuites/acl/test_eacl.py (hunk 4 above)
Comment (vvarg229) [reply pytest_tests/testsuites/acl/test_eacl.py:298]: > This doesn't make much sense.

Fixed

File: pytest_tests/testsuites/acl/test_eacl.py (hunk 5 above)
Comment (vvarg229) [reply pytest_tests/testsuites/acl/test_eacl.py:315]: > Like this

Fixed

File: pytest_tests/testsuites/acl/test_eacl.py (hunk 6 above)
Comment (vvarg229) [reply pytest_tests/testsuites/acl/test_eacl.py:332]: > And this.

Fixed

File: pytest_tests/testsuites/acl/test_eacl.py (hunk 7 above)
Comment (vvarg229) [reply pytest_tests/testsuites/acl/test_eacl.py:351]: > And this

Fixed

File: pytest_tests/testsuites/acl/test_eacl.py (hunk 9 above)
Comment (vvarg229) [reply pytest_tests/testsuites/acl/test_eacl.py:388]: > While deletion test has some meaning, none of them should be able to do that.

Fixed

File: pytest_tests/testsuites/acl/test_eacl.py (hunk 8 above)
Comment (vvarg229) [reply pytest_tests/testsuites/acl/test_eacl.py:369]: > And this

Fixed

**Review by roman-khimov — APPROVED** [review]
A bit too many commits than needed (at least 47be96fb6495c271f1c597de8fd37690d277d70c and 15be7ce9497f83dedc642e2b1c121a08efb51b53 can be squashed), but OK.

**Pull request merged**